Assets/Scripts/Entities/Entity.cs

```diff
@@ -22,7 +22,10 @@ public Vector2 position
     public abstract void Update();
 
     // This is where input is received
-    public abstract void HandleInstruction(Instruction instruction);
+    public virtual void HandleInstruction(Instruction instruction)
+    {
+        this.transform.position = GameManager.GetTransformForInstruction(instruction, transform.position);
```

Comment:
I've been moving things into `GameManager.cs` into `static` functions -- that helps avoid state issues.